roles/network_plugin/calico/tasks/install.yml

```diff
@@ -232,10 +232,20 @@
           }
 
     - name: Calico | Process calico network pool
-      set_fact:
-        _calico_pool: "{{ _calico_pool_cmd.stdout | from_json | combine(_calico_pool, recursive=True) }}"
       when:
         - _calico_pool_cmd is success
+      block:
+        - name: Calico | Get current calico network pool blocksize
+          set_fact:
+            _calico_blocksize: >
+              {
+                "spec": {
+                  "blockSize": {{ (_calico_pool_cmd.stdout | from_json).spec.blockSize }}
```

Comment:
Do we need also to merge the CIDR?  update the CIDR of the existing ippool is dangerous.

Reply:
I'd say it depends on whether we consider changing the cidr as part of the playbook a valid use-case, and what's the "default history" of the `cidr` attribute is (I came across this little inconvenience because the blockSize default changed apparently, because those cluster never had an explicit setting in their inventory for that ^^).

Changing the blockSize is definitely out of scope, because it's simply impossible, since it's immutable in the IPPool. The cidr, I don't know.

Overall, I would err on the side on keeping it that way, for minimal changes (which might be breaking someone workflow => can't be the case for `blockSize`).

Reply:
Yeah, I think that's how it went. The clusters were installed some years ago (not by me), and I think for some it was the calico default value for blocksize (26) and for others... well I don't really know ^^.